holopy/core/io/io.py

```diff
@@ -69,19 +70,20 @@ def load(inf):
     loaded_yaml = False
     # attempt to load a holopy yaml file
     try:
+        #TODO also want to load hdf5 here
         loaded = serialize.load(inf)
         loaded_yaml = True
+        return loaded
     except (serialize.ReaderError, UnicodeDecodeError):
-        pass
-        # If that fails, we go on and read images
-
-    if not loaded_yaml:
-        pass
-        #TODO load a tif with metadata
-        #TODO confirm tif has metadata
-        #TODO if not, raise exception referring user to load_image function.
-
-    return loaded
+        if os.path.splitext(inf)[1] in tiflist:
+            im = load_image(inf)
+            metadat = yaml.load(pilimage.open(inf).tag[270][0])
```

Comment:
Is there a reason you are calling this metadat? It looks kind of like you left off a character. I would probably either spell out metadata, or if you want to shorten, just call it meta here.